streams/src/test/java/org/apache/kafka/streams/processor/internals/tasks/DefaultTaskManagerTest.java

```diff
@@ -94,6 +103,114 @@ public void shouldAssignTaskThatCanBeProcessed() {
         assertNull(taskManager.assignNextTask(taskExecutor));
     }
 
+    private class AwaitingRunnable implements Runnable {
+        private final CountDownLatch awaitDone = new CountDownLatch(1);
+        private final AtomicBoolean shutdownRequested = new AtomicBoolean(false);
+        @Override
+        public void run() {
+            while (!shutdownRequested.get()) {
+                try {
+                    taskManager.awaitProcessableTasks();
+                } catch (final InterruptedException ignored) {
+                }
+                awaitDone.countDown();
+            }
+        }
+
+        public void shutdown() {
+            shutdownRequested.set(true);
+            taskManager.signalProcessableTasks();
+        }
+    }
+
+    @Test
+    public void shouldBlockOnAwait() throws InterruptedException {
+        final AwaitingRunnable awaitingRunnable = new AwaitingRunnable();
+        final Thread awaitingThread = new Thread(awaitingRunnable);
+        awaitingThread.start();
+
+        assertFalse(awaitingRunnable.awaitDone.await(100, TimeUnit.MILLISECONDS));
```

Comment:
Do you want to use the `VERIFICATION_TIMEOUT` here?

Reply:
Thanks, yes.